lib/saluki-components/src/sources/checks/python_scheduler.rs

```diff
@@ -0,0 +1,486 @@
+        // If we have explicit source code provided, we can definitely run it
+        if check.check_source_code.is_some() {
+            debug!("Check has explicit source code, can definitely run it");
+            return true;
+        }
+
+        // Try to find the check in possible import paths
+        let module_paths = self.possible_import_paths_for_module(check.check_request.name.as_str());
+        debug!(
+            "Checking if module '{}' can be run, found {} possible paths",
+            check.check_request.name,
+            module_paths.len()
+        );
+
+        if module_paths.is_empty() {
+            warn!("No import paths found for check: {}", check.check_request.name);
+            return false;
+        }
+
+        for module_name in module_paths {
+            debug!("Attempting to register check from import path: {}", module_name);
+            match std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
+                self.register_check_from_imports(&module_name)
+            })) {
+                Ok(handle_result) => {
+                    let can_run = handle_result.is_ok();
+                    info!(
+                        "Checking if py module '{name}' can be run. Result: {}",
+                        can_run,
+                        name = module_name,
+                    );
+                    if can_run {
+                        return true;
+                    }
+                }
+                Err(e) => {
+                    error!("Python panic during module check for {}: {:?}", module_name, e);
+                    // Continue trying other modules
+                }
+            }
+        }
+
+        false
+    }
+
+    // This function does 3 things
+    // 1. Registers the check source code and gets a handle
+    // 2. Starts a local task for each instance that
+    //    queues a run of the check every min_collection_interval_ms
+    // 3. Stores the handles in the running hashmap
+    fn run_check(&mut self, check: &RunnableCheckRequest) -> Result<(), GenericError> {
+        let check_handle = self.register_check(check)?;
+        let running_entry = self
+            .running
+            .entry(check.check_request.source.clone())
+            .or_insert((check_handle.clone(), Vec::new()));
+
+        info!(
+            "Running check {name} with {num_instances} instances",
+            name = check.check_request.name,
+            num_instances = check.check_request.instances.len()
+        );
+        for (idx, instance) in check.check_request.instances.iter().enumerate() {
+            let instance = instance.clone();
+            let init_config = check.check_request.init_config.clone();
+
+            let check_handle = check_handle.clone();
+            trace!("Spawning task for check instance {idx}");
+            self.tlm.check_instances_started.increment(1);
+            let handle = tokio::task::spawn(async move {
```

Comment:
This is where we _actually_ schedule a check to run. We simply use tokio to schedule a lightweight "task" (similar to goroutine) and avoid any interaction with OS threads.